streams/src/main/java/org/apache/kafka/streams/processor/AbstractNotifyingRestoreCallback.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.processor;
+
+/**
+ * Abstract implementation of the  {@link StateRestoreCallback} used for batch restoration operations.
+ *
+ * Includes default no-op methods of the {@link StateRestoreListener} {@link #onRestoreStart(String, long, long)},
+ * {@link #onBatchRestored(String, long, long)}, and {@link #onRestoreEnd(String, long)}.
+ */
+public abstract class AbstractNotifyingRestoreCallback implements StateRestoreCallback, StateRestoreListener {
```

Comment:
I did not catch it in the KIP wiki but.. the class names are a bit inconsistent here:

```
AbstractBatchingRestoreCallback
AbstractNotifyingRestoreCallback
```

Better be either

```
AbstractBatchingRestoreCallback
AbstractRestoreCallback
```

or 

```
AbstractNotifyingBatchingRestoreCallback
AbstractNotifyingRestoreCallback
```

WDYT?